src/server/hotspot_partition_calculator.cpp

```diff
@@ -98,5 +104,60 @@ void hotspot_partition_calculator::data_analyse()
     }
 }
 
+/*static*/ void
+hotspot_partition_calculator::send_hotkey_detect_request(const std::string &app_name,
+                                                         const int partition_index,
+                                                         const hotkey_detect_type hotkey_type,
+                                                         const hotkey_detect_action action)
+{
+    dsn::apps::hotkey_detect_request request;
+    request.type = (hotkey_type == hotkey_detect_type::WRITE_HOTKEY_DETECT)
+                       ? dsn::apps::hotkey_type::WRITE
+                       : dsn::apps::hotkey_type::READ;
+    request.action = (action == hotkey_detect_action::STOP_HOTKEY_DETECT)
+                         ? dsn::apps::hotkey_detect_action::STOP
+                         : dsn::apps::hotkey_detect_action::START;
```

Comment:
Why does it need a mapping from hotkey_detect_action to `dsn::apps::hotkey_detect_action`. Why don't use `dsn::apps::hotkey_detect_action` directly as your function argument?

```suggestion
                                                         const dsn::apps::hotkey_type hotkey_type,
                                                         const dsn::apps::hotkey_detect_action action)
{
    dsn::apps::hotkey_detect_request request;
    request.type = hotkey_type;
    request.action = action;
```

Reply:
in rrdb_types.h(automatic generated by rrdb.thrift):
```cpp
struct hotkey_type
{
    enum type
    {
        READ = 0,
        WRITE = 1
    };
};
```
cannot convert ‘const pegasus::server::hotkey_detect_type’ to ‘dsn::apps::hotkey_type::type’ in assignment

Reply:
Of course, I mean delete the redundant type `pegasus::server::hotkey_detect_type`. No idea why need it.